src/esp/bindings_js/bindings_js.cpp

```diff
@@ -237,5 +302,32 @@ EMSCRIPTEN_BINDINGS(habitat_sim_bindings_js) {
       .function("addAgentToNode",
                 em::select_overload<Agent::ptr(const AgentConfiguration&,
                                                scene::SceneNode&)>(
-                    &Simulator::addAgent));
+                    &Simulator::addAgent))
+      .function("getExistingObjectIDs", &Simulator::getExistingObjectIDs)
+      .function("setObjectMotionType", &Simulator::setObjectMotionType)
+      .function("getObjectMotionType", &Simulator::getObjectMotionType)
+      .function("addObject", &Simulator::addObject, em::allow_raw_pointers())
+      .function("addObjectByHandle", &Simulator::addObjectByHandle,
+                em::allow_raw_pointers())
+      .function("removeObject", &Simulator::removeObject)
+      .function("setTransformation", &Simulator::setTransformation)
+      .function("getTransformation", &Simulator::getTransformation)
+      .function("setTranslation", &Simulator::setTranslation)
+      .function("getTranslation", &Simulator::getTranslation)
+      .function("setRotation", &Simulator::setRotation)
+      .function("getRotation", &Simulator::getRotation)
+      .function("setObjectLightSetup", &Simulator::setObjectLightSetup)
+      .function("getLightSetup", &Simulator::getLightSetup)
+      .function("setLightSetup", &Simulator::setLightSetup)
+      .function("stepWorld", &Simulator::stepWorld)
+      .function("syncGrippedObject", &Simulator::syncGrippedObject)
+      .function("initOrUpdateCrossHairNode",
+                &Simulator::initOrUpdateCrossHairNode)
+      .function("recomputeNavMesh", &Simulator::recomputeNavMesh)
+      .function("findNearestObjectUnderCrosshair",
+                &Simulator::findNearestObjectUnderCrosshair)
+      .function("unproject", &Simulator::unproject)
+      .function("getAgentTransformation", &Simulator::getAgentTransformation)
+      .function("getAgentAbsoluteTranslation",
+                &Simulator::getAgentAbsoluteTranslation);
```

Comment:
All the getter/setters in this section really ought to be properties of the format 

`.property("x", &MyClass::getX, &MyClass::setX)`

Reply:
@jturner65 we can only use properties for getter setters which are operating on a variable in simulator class right? `setRotation` `getRotation`,etc can't be changed to properties right?

Reply:
Nope, you can use them for anything.  They are just syntactic sugar.